server/ast/discard.go

```diff
@@ -27,5 +28,55 @@ func nodeDiscard(node *tree.Discard) (vitess.Statement, error) {
 	if node == nil {
 		return nil, nil
 	}
-	return nil, fmt.Errorf("DISCARD is not yet supported")
+	if node.Mode != tree.DiscardModeAll {
+		return nil, fmt.Errorf("unhandled DISCARD mode: %v", node.Mode)
+	}
+
+	return vitess.InjectedStatement{
+		Statement: DiscardStatement{},
+	}, nil
+}
+
+// DiscardStatement is just a marker type, since all functionality is handled by the connection handler,
+// rather than the engine. It has to conform to the sql.ExecSourceRel interface to be used in the handler, but this
+// functionality is all unused.
+type DiscardStatement struct{}
```

Comment:
We should throw this in the `server/node` folder with the other `vitess.Injectable` nodes.